apps/prairielearn/src/lib/github.js

```diff
@@ -50,25 +50,30 @@ async function createRepoFromTemplateAsync(client, repo, template) {
 
   // The above call will complete before the repo itself is actually ready to use,
   // so poll for a bit until all the files are finally copied in.
-  let repo_up = false;
-  const poll_time_ms = 100;
-  while (!repo_up) {
+  //
+  // We'll try for up to 30 seconds, polling every second.
+  for (let i = 0; i < 30; i++) {
```

Comment:
Loops with error handling are hard... please check that this is implemented sensibly 🙃